### PR TITLE
s3: add option for multipart failiure behaviour

### PR DESCRIPTION
**Pull request by ajankovic**
This is needed for resuming uploads across different sessions.

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/ncw/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/ncw/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by ncw** [comment]
This code looks great!

I don't think rclone will resume the upload though? Unless you've got some more PRs up your sleeve :-)

**Comment by ajankovic** [comment]
Yes I have plans for it. Since this option is one of the prerequisites I would like to have it in as it's not intrusive too much. What do you think?

File: backend/s3/s3.go [hunk 1]
```diff
@@ -748,6 +748,17 @@ Use this only if v4 signatures don't work, eg pre Jewel/v10 CEPH.`,
 See: [AWS S3 Transfer acceleration](https://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration-examples.html)`,
 			Default:  false,
 			Advanced: true,
+		}, {
+			Name:     "leave_parts_on_error",
+			Provider: "AWS",
+			Help: `If true avoid calling abort upload on a failure, leaving all successfully uploaded parts on S3 for manual recovery.
+
+It should be set to true for resuming uploads across different sessions.
```
Comment (ncw) [inline backend/s3/s3.go]: Perhaps note that rclone can't do this yet?

Comment (ajankovic) [reply backend/s3/s3.go]: Or I should remove that line.

Comment (ncw) [reply backend/s3/s3.go]: Removing the line seems perfect

**Comment by ncw** [comment]
Oops forgot to merge this!  Thank you :-)

**Comment by MagTun** [comment]
I am confused by this thread and the help section of [S3](https://rclone.org/s3/#amazon-s3). 

I have read [here ](https://github.com/rclone/rclone/issues/87) that `you can't quit rclone and run it again to resume`.

But this `It should be set to true for resuming uploads across different sessions.` is really misleading because it implies `different sessions **of rclone**`: which I would interpret as `I can close rclone, switch off my computer, switch it on, run the same command from a fresh .cmd and pick up where the multipart upload was left off`. 

But this conflict with the `you can't quit rclone and run it again to resume`. 

Sure the help mentions the `manual recovery` but still the `different sessions` should be better explained: which different sessions are you talking about, rclone, another session in aws cli...

Would be great if you could clarify!

Thanks

**Comment by ncw** [comment]
> I am confused by this thread and the help section of [S3](https://rclone.org/s3/#amazon-s3).
> 
> I have read [here ](https://github.com/rclone/rclone/issues/87) that `you can't quit rclone and run it again to resume`.

That is correct. Rclone won't resume an upload if you quit it.

> But this `It should be set to true for resuming uploads across different sessions.` is really misleading because it implies `different sessions **of rclone**`: which I would interpret as `I can close rclone, switch off my computer, switch it on, run the same command from a fresh .cmd and pick up where the multipart upload was left off`.
> 
> But this conflict with the `you can't quit rclone and run it again to resume`.

This will be part of a future feature that will let you do that. You can't do it yet though.

> Sure the help mentions the `manual recovery` but still the `different sessions` should be better explained: which different sessions are you talking about, rclone, another session in aws cli...

At the moment it would be a different session with something else, but one day rclone will do this for you!

**Comment by MagTun** [comment]
Super, that's a great news! 

Any idea how far in the future this "one day" is? Is it a couple of days/weeks or months/years?

**Comment by ncw** [comment]
It is my TODO list but more likely months than weeks!

**Pull request merged**